tests/lib/rules/prop-types.js

```diff
@@ -2366,6 +2386,38 @@ ruleTester.run('prop-types', rule, {
       }]
     }, {
       code: [
+        'var Hello = function(props) {',
+        '  return <div>{props.firstname}</div>;',
+        '};',
+        'Hello.propTypes = {}'
+      ].join('\n'),
+      options: [{skipUndeclared: true}],
+      parserOptions: parserOptions,
+      errors: [{
+        message: '\'firstname\' is missing in props validation',
+        line: 2,
+        column: 22
+      }]
+    }, {
+      code: [
+        'class Hello extends React.Component {',
+        '  static get propTypes() {',
```

Comment:
Let's not worry about consolidating them in this PR, but please do add a `Hello.propTypes = { … }` variant.